api_server.go

```diff
@@ -278,13 +280,138 @@ func (x *ApiServer) SheetsUpload(w http.ResponseWriter, r *http.Request) {
 		}
 
 	default:
-		http.Error(w, "", http.StatusMethodNotAllowed)
+		methodNotAllowed(w)
+	}
+}
+
+type UploadType string
```

Comment:
Namespacing could get weird here. Might be time to start splitting out into separate packages.

Reply:
esp if we decide on protobuf.